File: CS/BingElevation/Form1.cs
using DevExpress.XtraMap;
using System;
using System.Windows.Forms;

namespace BingElevation {
    public partial class Form1 : Form {
        const string yourBingKey = "Enter Your Bing Key Here";
        public Form1() {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e) {
            imageDataProvider.BingKey = yourBingKey;

            #region #ElevationDataProviderConfiguring
            // Create a new elevation data provider.
            BingElevationDataProvider dataProvider = new BingElevationDataProvider {
                BingKey = yourBingKey,
                ProcessMouseEvents = true
            };
            informationLayer.DataProvider = dataProvider;
            // Process result obtaining.
            dataProvider.ElevationsCalculated += OnElevationsCalculated;
            #endregion #ElevationDataProviderConfiguring
        }

        #region #ElevationObtained
        private void OnElevationsCalculated(object sender, ElevationsCalculatedEventArgs e) {
            if(e.Cancelled == true || e.Error != null) return;
            if(e.Result.ResultCode != RequestResultCode.Success) return;

            informationLayer.Data.Items.Clear();
            // foreach requested location create a new callout with elevation value.
            foreach (ElevationInformation elevationResult in e.Result.Locations) {
                informationLayer.Data.Items.Add(new MapCallout {
                    Location = elevationResult.Location,
                    Text = String.Format("Elevation: {0}m.", elevationResult)
                });
            }
        }
        #endregion #ElevationObtained
    }
}

File: VB/BingElevation/Form1.vb
Imports DevExpress.XtraMap
Imports System
Imports System.Windows.Forms

Namespace BingElevation
    Partial Public Class Form1
        Inherits Form

        Private Const yourBingKey As String = "Enter Your Bing Key Here"
        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
            imageDataProvider.BingKey = yourBingKey

'            #Region "#ElevationDataProviderConfiguring"
            ' Create a new elevation data provider.
            Dim dataProvider As BingElevationDataProvider = New BingElevationDataProvider With {.BingKey = yourBingKey, .ProcessMouseEvents = True}
            informationLayer.DataProvider = dataProvider
            ' Process result obtaining.
            AddHandler dataProvider.ElevationsCalculated, AddressOf OnElevationsCalculated
'            #End Region ' #ElevationDataProviderConfiguring
        End Sub

        #Region "#ElevationObtained"
        Private Sub OnElevationsCalculated(ByVal sender As Object, ByVal e As ElevationsCalculatedEventArgs)
            If e.Cancelled = True OrElse e.Error IsNot Nothing Then
                Return
            End If
            If e.Result.ResultCode <> RequestResultCode.Success Then
                Return
            End If

            informationLayer.Data.Items.Clear()
            ' foreach requested location create a new callout with elevation value.
            For Each elevationResult As ElevationInformation In e.Result.Locations
                informationLayer.Data.Items.Add(New MapCallout With {.Location = elevationResult.Location, .Text = String.Format("Elevation: {0}m.", elevationResult)})
            Next elevationResult
        End Sub
        #End Region ' #ElevationObtained
    End Class
End Namespace
